neurolib/models/multimodel/builder/adex.py

```diff
@@ -0,0 +1,944 @@
+            linear-nonlinear cascade for inhibitory AdEx mass, if None, will
+            look for it in this directory
+        :type inh_lin_nonlin_cascade_filename: list[str]|str|None
+        :param local_connectivity: local within-node connectivity matrix
+        :type local_connectivity: np.ndarray
+        :param local_delays: local within-node delay matrix
+        :type local_delays: list[np.ndarray]|np.ndarray
+        :param exc_seed: seed for random number generator for the excitatory
+            masses
+        :type exc_seed: int|None
+        :param inh_seed: seed for random number generator for the excitatory
+            masses
+        :type inh_seed: int|None
+        """
+        num_nodes = connectivity_matrix.shape[0]
+        exc_mass_params = self._prepare_mass_params(exc_mass_params, num_nodes)
+        inh_mass_params = self._prepare_mass_params(inh_mass_params, num_nodes)
+        exc_lin_nonlin_cascade_filename = self._prepare_mass_params(
+            exc_lin_nonlin_cascade_filename, num_nodes, native_type=str
+        )
+        inh_lin_nonlin_cascade_filename = self._prepare_mass_params(
+            inh_lin_nonlin_cascade_filename, num_nodes, native_type=str
+        )
+        local_connectivity = self._prepare_mass_params(local_connectivity, num_nodes, native_type=np.ndarray)
+        local_delays = self._prepare_mass_params(local_delays, num_nodes, native_type=np.ndarray)
+        exc_seeds = self._prepare_mass_params(exc_seed, num_nodes, native_type=int)
+        inh_seeds = self._prepare_mass_params(inh_seed, num_nodes, native_type=int)
+
+        nodes = []
+        for (i, (exc_params, inh_params, exc_cascade, inh_cascade, local_conn, local_dels,),) in enumerate(
+            zip(
+                exc_mass_params,
+                inh_mass_params,
+                exc_lin_nonlin_cascade_filename,
+                inh_lin_nonlin_cascade_filename,
+                local_connectivity,
+                local_delays,
+            )
+        ):
+            node = AdExNode(
+                exc_params=exc_params,
+                inh_params=inh_params,
+                exc_lin_nonlin_cascade_filename=exc_cascade,
+                inh_lin_nonlin_cascade_filename=inh_cascade,
+                connectivity=local_conn,
+                delays=local_dels,
+                exc_seed=exc_seeds[i],
+                inh_seed=inh_seeds[i],
+            )
+            node.index = i
+            node.idx_state_var = i * node.num_state_variables
+            # set correct indices of noise input
+            for mass in node:
+                mass.noise_input_idx = [2 * i + mass.index]
+            nodes.append(node)
+
+        super().__init__(
+            nodes=nodes, connectivity_matrix=connectivity_matrix, delay_matrix=delay_matrix,
+        )
+        # assert we have 2 sync variable
+        assert len(self.sync_variables) == 2
+
+    def _sync(self):
```

Comment:
unfortunately - due to squared coupling it's just easier to redefine `network._sync()` rather than trying to accommodate all version - then the superclass `Network` would be too complicated... so I'm redefining here to accommodate network couplings - both basic and squared variants are computed here